plugin/sqlsh/src/main/java/com/twosigma/beaker/sqlsh/utils/QueryExecutor.java

```diff
@@ -153,7 +153,7 @@ protected static String setObject(String sql, Object value){
       String begin = sql.substring(0, index);
       String end = sql.substring(index + 1, sql.length());
       if(value instanceof String){
-        ret = begin + "\'" + value + "\'" + end;
+        ret = begin + "'" + replaseString(((String) value), "'", "''", 0) + "'" + end;
```

Comment:
Yes there is replaceAll method in Java. But I tried different escape
characters. And for some it was difficult to write regular expressions.
That's why I wrote custom method. If for some reason we change escape
character  (for some DB), it will be easy. Let me know if I need to rewrite
this part.

3 Авг 2016 г. 7:03 PM пользователь "Scott Draves" <notifications@github.com>
написал:

> In
> plugin/sqlsh/src/main/java/com/twosigma/beaker/sqlsh/utils/QueryExecutor.java
> <https://github.com/twosigma/beaker-notebook/pull/4591#discussion_r73344038>
> :
>
> > @@ -153,7 +153,7 @@ protected static String setObject(String sql, Object value){
> >        String begin = sql.substring(0, index);
> >        String end = sql.substring(index + 1, sql.length());
> >        if(value instanceof String){
> > -        ret = begin + "\'" + value + "\'" + end;
> > +        ret = begin + "'" + replaseString(((String) value), "'", "''", 0) + "'" + end;
>
> doesn't Java have replaceAll() method?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/twosigma/beaker-notebook/pull/4591/files/bb1edbf8c66b097c661e46bc6be42e28b744ffd6#r73344038>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKH3afxVi-0vsxxso8TcWwcde3vw9rPeks5qcKfqgaJpZM4JbcC8>
> .
>